service/modelmanager.go

```diff
@@ -0,0 +1,71 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016-2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package service
+
+import "log"
+
+const createModelTableSQL = `
+	CREATE TABLE IF NOT EXISTS model (
+		id          serial primary key not null,
+		brand_id    varchar(200) not null,
+		name        varchar(200) not null,
+		signing_key text default '',
+		revision    int
+	)
+`
+const listModelsSQL = "select id, brand_id, name, signing_key, revision from model order by name"
+
+// Model holds the model details in the local database
+type Model struct {
+	ID         int
+	BrandID    string
+	Name       string
+	SigningKey string
+	Revision   int
+}
+
+// CreateModelTable creates the database table for a model.
+func (db *DB) CreateModelTable() error {
+	_, err := db.Exec(createModelTableSQL)
+	return err
+}
+
+// ListModels fetches the full catalogue of models from the database.
+func (db *DB) ListModels() ([]Model, error) {
+	var models []Model
+
+	rows, err := db.Query(listModelsSQL)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
This shouldn't be a Fatal error, as this is called from a HTTP handler, so if there is an error, the server will shutdown. Instead of that, it should return an error, so the handler can return an error to the user gracefully